crates/datasources/Cargo.toml

```diff
@@ -13,6 +13,7 @@ apache-avro = "0.16"
 async-channel = "2.1.1"
 async-stream = "0.3.5"
 async-trait = { workspace = true }
+arrow-schema = { version = "50.0.0", default-features = false }
```

Comment:
in general we should use the version of arrow that's reexported from `datafusion` at `datafusion::arrow`

Reply:
Yeah I was having some difficulties using the downcast_dictionary_array! macro in write/demux.rs using the re-export.  I'm not sure how to get around this issue yet... will try to find a solution.

```
error[E0433]: failed to resolve: use of undeclared crate or module `arrow_schema`
   --> crates/datasources/src/common/sink/write/demux.rs:344:17
    |
344 | / ...   downcast_dictionary_array!(
345 | | ...       col_array =>  {
346 | | ...           let array = col_array.downcast_dict::<StringArray>()
347 | | ...               .ok_or(DataFusionError::Execution(format!("it is not yet supported to write to hive partitions with datatype {}",
...   |
356 | | ...       _ => unreachable!(),
357 | | ...   )
    | |_______^ use of undeclared crate or module `arrow_schema`
    |
    = note: this error originates in the macro `downcast_dictionary_array` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
    |
29  + use calamine::DataType;
    |
29  + use deltalake::kernel::DataType;
    |

```

Reply:
I think it may be because it expands to code that includes the `arrow_schema` qualifier?

```
#[macro_export]
macro_rules! downcast_dictionary_array {
    ($values:ident => $e:expr, $($p:pat => $fallback:expr $(,)*)*) => {
        downcast_dictionary_array!($values => {$e} $($p => $fallback)*)
    };

    ($values:ident => $e:block $($p:pat => $fallback:expr $(,)*)*) => {
        match $values.data_type() {
            arrow_schema::DataType::Dictionary(k, _) => {
                $crate::downcast_integer! {
                    k.as_ref() => ($crate::downcast_dictionary_array_helper, $values, $e),
                    k => unreachable!("unsupported dictionary key type: {}", k)
                }
            }
            $($p => $fallback,)*
        }
    }
}
```